tests/UnitTests/CoreGraphics.drawing/CGContextDrawing_ClearRectTests.cpp

```diff
@@ -0,0 +1,102 @@
+//******************************************************************************
+//
+// Copyright (c) Microsoft. All rights reserved.
+//
+// This code is licensed under the MIT License (MIT).
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//******************************************************************************
+
+#include "DrawingTest.h"
+
+#pragma region ClearRect
+
+class CGClearRect : public WhiteBackgroundTest<>, public ::testing::WithParamInterface<::testing::tuple<CGRect, CGAffineTransform>> {
+    CFStringRef CreateOutputFilename() {
+        CGRect rect = ::testing::get<0>(GetParam());
+        CGAffineTransform transformation = ::testing::get<1>(GetParam());
+
+        return CFStringCreateWithFormat(nullptr,
+                                        nullptr,
+                                        CFSTR("TestImage.CGContext.ClearRect.rect.(%0.0f.%0.0f)%0.0fx%0.0f.transformationAffine.[%0.1f,%0."
+                                              "1f,%0.1f,%0.1f,%0.1f,%0.1f].png"),
+                                        rect.origin.x,
+                                        rect.origin.y,
+                                        rect.size.width,
+                                        rect.size.height,
+                                        transformation.a,
+                                        transformation.b,
+                                        transformation.c,
+                                        transformation.d,
+                                        transformation.tx,
+                                        transformation.ty);
+    }
+};
+
+TEST_P(CGClearRect, Transformed) {
+    CGRect rect = ::testing::get<0>(GetParam());
+    CGAffineTransform transformation = ::testing::get<1>(GetParam());
+
+    CGContextRef context = GetDrawingContext();
+
+    CGContextAddRect(context, CGRectMake(100, 50, 50, 50));
+    CGContextConcatCTM(context, transformation);
+    CGContextClip(context);
+
+    CGContextSetRGBFillColor(context, 0.48, 0.73, 0, 1);
+    CGContextClearRect(context, rect);
+}
+
+TEST_P(CGClearRect, Transformed2) {
+    CGRect rect = ::testing::get<0>(GetParam());
+    CGAffineTransform transformation = ::testing::get<1>(GetParam());
+
+    CGContextRef context = GetDrawingContext();
+    CGRect bounds = GetDrawingBounds();
+
+    CGContextAddRect(context, CGRectMake(0, 0, 250, 250));
+    CGContextConcatCTM(context, transformation);
+    CGContextClip(context);
```

Comment:
yes, i would of made both of them parameterized but then we would end up with a spew of files.

---
In reply to: [102640945](https://github.com/Microsoft/WinObjC/pull/2053#discussion_r102640945) [](ancestors = 102640945)